src/clarity-angular/emphasis/alert/alerts.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+   AfterContentInit,
+   Component,
+   ContentChildren,
+   EventEmitter,
+   Input,
+   Output,
+   QueryList
+} from "@angular/core";
+import {Alert} from "./alert";
+import {MultiAlertService} from "./providers/multi-alert-service";
+
+@Component({
+   selector: "clr-alerts",
+   templateUrl: "./alerts.html",
+   providers: [MultiAlertService],
+   host: {
+      "[class.alerts]": "true",
+      "[class.alert-danger]": "this.currentAlertType == 'danger'",
+      "[class.alert-info]": "this.currentAlertType == 'info'",
+      "[class.alert-success]": "this.currentAlertType == 'success'",
+      "[class.alert-warning]": "this.currentAlertType == 'warning'"
+   }
+})
+export class Alerts implements AfterContentInit {
+   @ContentChildren(Alert) allAlerts: QueryList<Alert>;
+
+   /**
+    * Input/Output to support two way binding on current alert index
+    */
+   @Input("clrCurrentAlertIndex") public set _inputCurrentIndex(index: number) {
+      if (index) {
+         this.multiAlertService.current = index;
+      }
+   }
+
+   @Output("clrCurrentAlertIndexChange") public currentAlertIndexChange = new EventEmitter<number>(false);
```

Comment:
I'm surprised, is this output actually fired when the alert changes because the user clicks on the pager?
I see the pager updating the service, but this component doesn't subscribe to the service.

What I would do is subscribe to the service, and fire the output when the service changes. It would even remove a lot of safety guards from this class.

Reply:
Updated.